benchmarks/algorithm/rotate.cpp

```diff
@@ -0,0 +1,73 @@
+#include <nanorange/algorithm/rotate.hpp>
+
+#include <algorithm>
+#include <utility>
+#include <vector>
+
+#include <nanorange/algorithm/generate.hpp>
```

Comment:
This file calls `std::generate` but not to `nano::rotate`; I think this include is unused.

Reply:
Done - used nano::generate. I fell back to the std::generate since I was getting some clang-format issues.
Any thoughts on the problems listed in the issue itself?